packages/protocol/contracts/dapis/RrpDapiServer.sol

```diff
@@ -0,0 +1,178 @@
+      dapis[dapiId].requestIndexResetter = requestIndexResetter;
+      dapis[dapiId].nextRequestIndex = 1;
+    }
+
+    function resetDapiRequestIndex(bytes16 dapiId)
+        external
+    {
+        Dapi storage dapi = dapis[dapiId];
+        require(
+            msg.sender == dapi.requestIndexResetter,
+            "Caller not resetter"
+            );
+        dapi.nextRequestIndex = 1;
+        dapi.requestIndexResetCount++;
+    }
+
+    function makeDapiRequest(
+        bytes16 dapiId,
+        bytes calldata parameters
+        )
+        external
+        returns (uint64 currDapiRequestIndex)
+    {
+        Dapi storage dapi = dapis[dapiId];
+        currDapiRequestIndex = dapi.nextRequestIndex++;
+        dapi.requestIndexToResponsesLength[currDapiRequestIndex] = 0;
+        if (dapi.requestIndexToResponses[currDapiRequestIndex].length == 0) {
+            dapi.requestIndexToResponses[currDapiRequestIndex] = new int256[](dapi.noResponsesToReduce);
+        }
+        require(
+            airnodeRrp.requesterIndexToClientAddressToEndorsementStatus(dapi.requesterIndex, msg.sender),
+            "Caller not endorsed"
+            );
+        for (uint i = 0; i < dapi.templateIds.length; i++) {
+            (bool success, bytes memory returnedData) = address(airnodeRrp).delegatecall(abi.encodeWithSelector( // solhint-disable-line
+                AirnodeRrp.makeRequest.selector,
+                dapi.templateIds[i],
+                dapi.requesterIndex,
+                dapi.designatedWallets[i],
+                address(this),
+                this.fulfill.selector,
+                parameters
+                ));
+            require(success, "Request unsuccessful"); // This will never happen if AirnodeRrp is valid
+            bytes32 requestId = abi.decode(returnedData, (bytes32));
+            airnodeRequestIdToDapiRequestIdentifiers[requestId] = DapiRequestIdentifiers({
+                dapiId: dapiId,
+                dapiRequestIndex: uint64(currDapiRequestIndex),
+                dapiRequestIndexResetCountAtRequestTime: dapi.requestIndexResetCount
+                });
+        }
+    }
+
+    function fulfill(
+        bytes32 requestId,
+        uint256 statusCode,
+        bytes calldata data
+        )
+        external
+    {
+        require(
+            address(airnodeRrp) == msg.sender,
+            "Caller not AirnodeRrp"
+            );
+        DapiRequestIdentifiers storage dapiRequestIdentifiers = airnodeRequestIdToDapiRequestIdentifiers[requestId]; 
+        require(
+            dapiRequestIdentifiers.dapiRequestIndex != 0,
+            "Request ID invalid"
+            );
+        if (statusCode == 0) {
+            Dapi storage dapi = dapis[dapiRequestIdentifiers.dapiId];
+            require(
+                dapiRequestIdentifiers.dapiRequestIndexResetCountAtRequestTime == dapi.requestIndexResetCount,
+                "Request stale"
+                );
```

Comment:
dAPI owner has reset the dAPI request indices after this dAPI request has been made